src/domain/categories.ts

```diff
@@ -45,46 +42,81 @@ export function editCategory(props: Category) {
 
 //ini namanya custom hooks
 export function useFetchCategories() {
-  const [categories, setCategories] = useState<Category[]>();
+  const [categories, setCategories] = useState<Category[]>([]);
+  const [errorMessageFetch, seterrorMessageFetch] = useState<string>("");
   const [isLoadingFetch, setIsLoadingFetch] = useState(false);
   useEffect(() => {
-    fetchData().then((categories) => {
-      setCategories(categories);
-    });
+    setIsLoadingFetch(true);
+    fetchData()
+      .then((data: Category[]) => {
+        setCategories([...data]);
+        setIsLoadingFetch(false);
+      })
+      .catch((err) => {
+        setCategories([]);
+        seterrorMessageFetch(`${err.message} + error in get data`);
+        setIsLoadingFetch(false);
+      });
   }, []);
   function reFetch() {
-    fetchData().then((categories) => {
-      setCategories(categories);
-    });
+    setIsLoadingFetch(true);
+    fetchData()
+      .then((data: Category[]) => {
+        setCategories([...data]);
+        setIsLoadingFetch(false);
+      })
+      .catch((err) => {
+        setCategories([]);
+        seterrorMessageFetch(`${err.message} + error in get data`);
+        setIsLoadingFetch(false);
+      });
   }
-  return { categories, reFetch, isLoadingFetch };
+  return { categories, errorMessageFetch, reFetch, isLoadingFetch };
 }
 
 export function useCreateCategory() {
   const [isLoadingCreate, setIsLoadingCreate] = useState(false);
-  function submit(props: Category) {
+  const [errorMessageCreate, seterrorMessageCreate] = useState("");
+  function submit(props: CategoryRequest) {
     setIsLoadingCreate(true);
-    return postCategory(props).then(() => {
-      setIsLoadingCreate(false);
-    });
+    return postCategory(props)
+      .then(() => {
+        setIsLoadingCreate(false);
+      })
+      .catch((err) => {
+        seterrorMessageCreate(`${err.message} + error in create`);
+        setIsLoadingCreate(false);
+      });
```

Comment:
Errornya jangan lupa diclear ketika sukses : 

```suggestion
    return postCategory(props)
      .then(() => {
        setErrorMessageCreate("");
      })
      .catch((err) => {
        setErrorMessageCreate(`${err.message} + error in create`);
      })
      .finally(() => setIsLoadingCreate(false));
```

Reply:
updated mas